src/test/java/api/loans/CheckInByBarcodeTests.java

```diff
@@ -403,18 +407,16 @@ public void shouldNotSendPatronNoticeWhenCheckInNoticeIsDefinedAndCheckInDoesNot
     final IndividualResource nod = itemsFixture.basedUponNod(
       builder -> builder.withTemporaryLocation(homeLocation.getId()));
 
-    loansFixture.checkInByBarcode(
-      nod, new DateTime(2018, 3, 5, 14, 23, 41, DateTimeZone.UTC),
-      checkInServicePointId);
     final CheckInByBarcodeResponse checkInResponse = loansFixture.checkInByBarcode(
       nod, new DateTime(2018, 3, 5, 14, 23, 41, DateTimeZone.UTC),
       checkInServicePointId);
 
     assertThat("Response should not include a loan",
       checkInResponse.getJson().containsKey("loan"), is(false));
 
+    Thread.sleep(5000);
```

Comment:
How long do we think it takes to trigger the request to the patron notice API?

Using 5 seconds across the board means that this test will *always* take at least 5 seconds to execute (which is quite a lot of build time).

Reply:
Talking about the time it takes to save notice in FakeStorageModule during tests, it takes at about 50ms. Agree, 5 second may to much.

Reply:
One more point about suing _Thread.sleep()_: sonar marks it as an issue.

Reply:
Sonar does not treat _TimeUnit.SECONDS.sleep(1)_ as and issue. It can be a workaround. The question is how much time we should wait. Do you think 1 second is enough? 
Even if sending notice takes more than 1 sec, test won't fail. (As we expect it not to be sent)

Reply:
Thanks for finding a solution that Sonar is happier with.

I think the key is using the same time out for both the positive (presence of a notice) and negative (absence) scenarios.

We can start with 1 second and increase it if it results in flaky tests. It is probably better to start smaller and increase, than start with a large delay, as that is unlikely to produce much feedback.

Reply:
We evaluate tests in Sonar? That's news to me. I would think, who cares what we put in the tests in order to get the job done. I don't recall sonar ever complaining about my crappy unit tests. Hmm...